docs/development/design/rate-limiting.md

```diff
@@ -0,0 +1,37 @@
+# Used Envoy for Rate Limiting
+
+<!-- [Issue #600](https://github.com/redhat-et/issues/600) -->
+
+## Summary
+
+To protect the apiserver service from a misbehaving tenant, the rest APIs should be rate limited. This will allow us to better reason about how to scale the service to a large number of tenants as there will be limit to the ammout of apiserver resources each tenant can consume.
+
+This also translates to a limit of number of devices that a tenant can enroll before he starts to hit rate limits.  There should be a way to increase the limits per tenant.
```

Comment:
>  There should be a way to increase the limits per tenant.

How does your proposal allow this? Limitador config is file based, and can't be adjusted at runtime as far as I can tell. Adjusting files - which are part of the deployment - isn't a great solution. Ideally the rate-limit config could be read from the database.

Reply:
Maybe we could put some info into the JWT and use that to pick the right rate limiting config for the tenant.

Reply:
That would require moving user/organization membership into the ID Provider though 🤔 
It's something I pondered on when writing the AuthZ design, but not sure that it's viable for us.

Reply:
Don't think it has to be the user/org in the IDP.  it could be a tier level attribute assigned to the user like: free, bronze, gold, etc.

Reply:
Even so, someone needs to assign that attribute in the IdP and that can't be done by OIDC.

Reply:
Passing jwt claims to limitador is possible